clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -750,15 +788,31 @@ public Node leastLoadedNode(long now) {
         if (algo == null) {
             throw new IllegalStateException("leastLoadedNodeAlgorithm cannot be null");
         }
+        List<Node> newNodes;
+        if (this.metadataUpdater.isUpdateClusterMetadataDue(now)) {
+            //update cluster metadata due, resolve bootstrap server and use the new resolved
+            //nodes for least loaded node selection
+            List<InetSocketAddress> addresses = ClientUtils.parseAndValidateAddresses(
+                this.bootstrapServers,
+                this.clientDnsLookup.toString());
+            newNodes = new ArrayList<>();
+
+            int nodeId = -1;
+            for (InetSocketAddress address : addresses) {
+                newNodes.add(new Node(nodeId--, address.getHostString(), address.getPort()));
+            }
+        } else {
+            newNodes = nodes;
+        }
```

Comment:
should we skip the algorithm selection?  maybe just use a random node. 
since those are fake nodes (with negative IDs),  algorithm selection won't help here?

Reply:
makes sense, modified.